src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7273,6 +7375,11 @@ private int GetReceiverSlotForMemberPostConditions(MethodSymbol? method)
         private void ApplyMemberPostConditions(int receiverSlot, MethodSymbol method)
         {
             Debug.Assert(receiverSlot >= 0);
+            if (method.GetIsNewExtensionMember())
+            {
+                return;
```

Comment:
Why are post conditions skipped for new extension members?

Reply:
The are not just "post-conditions", they are "member post-conditions". The attributes that are processed below are `MemberNotNull` and `MemberNotNullWhen`.
Normally those apply to members of the containing type, which holds states. My thinking was that extensions don't hold state, so those attributes don't apply.
But extension members could be stateful, so it may make sense after all.
I'll add a follow-up comment to confirm whether we want this to work. If so, we'll need to spec it first. Thanks!